src/main/java/io/aiven/kafka/connect/opensearch/RecordConverter.java

```diff
@@ -60,16 +59,26 @@ public class RecordConverter {
 
     private final OpensearchSinkConnectorConfig config;
 
+    private final DocumentIDStrategy globalDocIdGenerator;
+    private final DocumentIDStrategy perTopicDocIdGenerator; // Used on topics present in TOPIC_KEY_IGNORE_CONFIG
+
     public RecordConverter(final Boolean useCompactMapEntries,
                            final RecordConverter.BehaviorOnNullValues behaviorOnNullValues) {
         this(null);
     }
 
     public RecordConverter(final OpensearchSinkConnectorConfig config) {
         this.config = config;
+        globalDocIdGenerator = (config != null && config.ignoreKey()) ? config.docIdStrategy()
```

Comment:
https://github.com/aiven/opensearch-connector-for-apache-kafka/pull/139#discussion_r1024447961 , the `RecordConverter` should not make the decision what docId strategy to use - it has to be provided by configuration